src/internal/pfsdb/branches_test.go

```diff
@@ -312,3 +312,62 @@ func TestDeleteBranch(t *testing.T) {
 		})
 	})
 }
+
+func TestBranchTrigger(t *testing.T) {
+	t.Parallel()
+	withDB(t, func(ctx context.Context, t *testing.T, db *pachsql.DB) {
+		var masterBranchID, stagingBranchID pfsdb.BranchID
+		// Create two branches, master and staging in the same repo.
+		withTx(t, ctx, db, func(ctx context.Context, tx *pachsql.Tx) {
+			var err error
+			repoInfo := newRepoInfo(&pfs.Project{Name: "project1"}, "repo1", pfs.UserRepoType)
+			commit1 := createCommitPair(t, ctx, tx, newCommitInfo(repoInfo.Repo, random.String(32), nil)).CommitInfo.Commit
```

Comment:
I renamed `CommitPair` to `CommitWithID` since I added the revision information there and also to follow up on Jon's comment.